script/production/ConfigWhitelistOracleAdapters.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# Check if the file path is provided
+if [ -z "$1" ]; then
+  echo "Usage: $0 <csv_file_path>"
+  exit 1
+fi
+
+csv_file="$1"
+
+if ! script/utils/checkEnvironment.sh; then
+    echo "Environment check failed. Exiting."
+    exit 1
+fi
+
+source .env
+
+read -p "Enter the Adapter Registry address: " adapter_registry
+read -p "Do you want to perform the operation in a batch? (y/n) (default: y): " is_batch
+is_batch=${is_batch:-y}
+
+if [ "$is_batch" == "y" ]; then
+    read -p "Enter the EVC address: " evc
+fi
+
+onBehalfOf=$(cast wallet address --private-key $DEPLOYER_KEY)
+items="["
+
+while IFS=, read -r -a columns || [ -n "$columns" ]; do
+    adapterName="${columns[3]}"
+    adapter="${columns[4]}"
+    base="${columns[5]}"
+    quote="${columns[6]}"
+    whitelist="${columns[7]}"
+
+    if [[ "$adapter" == "Adapter" ]]; then
+        continue
+    fi
+
+    entry=$(cast call $adapter_registry "entries(address)((uint128,uint128))" $adapter --rpc-url $DEPLOYMENT_RPC_URL)
+    addedAt=$(echo $entry | cut -d'(' -f2 | cut -d',' -f1)
+    revokedAt=$(echo $entry | cut -d',' -f2 | cut -d')' -f1 | tr -d ' ')
+
+    if [[ "$whitelist" == "Yes" ]]; then
+        if [[ "$base" == "" || "$quote" == "" ]]; then
+            echo "No base or quote address found for the adapter $adapter. Skipping..."
+            continue
+        fi
+
+        if [[ $addedAt == "0" && $revokedAt == "0" ]]; then
+            if [ "$is_batch" != "y" ]; then
```

Comment:
Let's just always assume we're doing it in a batch, no? One fewer rarely-used code-path to maintain.